src/Graphics.jl

```diff
@@ -672,8 +682,12 @@ source color.
 @mustimplement set_source(gc::GraphicsContext, src)
 
 function set_source(gc::GraphicsContext, c::Color)
-    rgb = convert(RGB, c)
-    set_source_rgb(gc, rgb.r, rgb.g, rgb.b)
+    c isa AbstractRGB && return set_source_rgb(gc, red(c), green(c), blue(c))
```

Comment:
Maybe slightly cleaner with dispatch but no big deal either way

Reply:
I almost forgot the original intent, but the specialization can introduce ambiguity in the backend implementations.